models/classes/Tool/Validation/Lti1p3Validator.php

```diff
@@ -90,6 +102,17 @@ public function validateRole(LtiMessagePayloadInterface $ltiMessagePayload): voi
 
     private function getRegistrationRepository(): RegistrationRepositoryInterface
     {
-        return $this->getServiceLocator()->getContainer()->get(RegistrationRepositoryInterface::class);
+        return $this->registrationRepository;
+    }
+
+    /**
+     * @return ToolLaunchValidator
+     */
+    private function getToolLaunchValidator(): AbstractLaunchValidator
+    {
+        return $this->tooLaunchValidator ?? new ToolLaunchValidator(
+            $this->getRegistrationRepository(),
```

Comment:
It seems that you can manage it within a service provider

Reply:
no, I can't.